src/spikeinterface/preprocessing/tests/test_astype.py

```diff
@@ -0,0 +1,42 @@
+import pytest
+from pathlib import Path
+import shutil
+
+from spikeinterface import set_global_tmp_folder, NumpyRecording
+from spikeinterface.core import generate_recording
+
+from spikeinterface.preprocessing import astype
+
+import numpy as np
+
+
+if hasattr(pytest, "global_test_folder"):
+    cache_folder = pytest.global_test_folder / "preprocessing"
+else:
+    cache_folder = Path("cache_folder") / "preprocessing"
+
+set_global_tmp_folder(cache_folder)
+
+
+def test_astype():
+    traces = (np.random.rand(10000, 4) * 100).astype("float32")
```

Comment:
I think these tests will work better with a user defined traces

Just write something like [1.0 ,3.0, 5.0, ...] or at least let's make them non-deterministic.

Reply:
I think it's ok to leave randomness here. It should work in any case since it's only casting the dtypes

Reply:
I think it is unlikely to create a problem here but I also don't see why the randomness is required. If it is meant to work as a test then it should be reproducible so you can debug it if it fails. But here, the failure is unlikely to come from randonmess so not a big concern ...

Anyway, not a big deal:
"A foolish consistency is the hobgoblin of little minds"

Reply:
I'll set a seed ;)

Reply:
done